lua/neogit/buffers/status/actions.lua

```diff
@@ -507,6 +507,25 @@ M.n_toggle = function(self)
   end
 end
 
+---@param self StatusBuffer
+M.n_open_fold = function(self)
+  return function()
+    local fold = self.buffer.ui:get_fold_under_cursor()
```

Comment:
Is this where the lazy loading of the diff is happening?

Reply:
Yes, it's the `fold.options.on_open(fold, self.buffer.ui)` function that does this.

defined here:
- https://github.com/NeogitOrg/neogit/blob/b3d69c2f5b729cc3574691864c503da6b66e3d79/lua/neogit/buffers/status/ui.lua#L303
- https://github.com/NeogitOrg/neogit/blob/b3d69c2f5b729cc3574691864c503da6b66e3d79/lua/neogit/buffers/status/ui.lua#L212-L243